FirebaseRemoteConfig/SwiftNew/RemoteConfig.swift

```diff
@@ -0,0 +1,961 @@
+        }
+        if let config = configContent.getConfigAndMetadata(forNamespace: fullyQualifiedNamespace)
+          as? [String: RemoteConfigValue] {
+          callListeners(key: key, config: config)
+        }
+        return
+      }
+
+      value = defaultValue(forFullyQualifiedNamespace: fullyQualifiedNamespace, key: key)
+    }
+    return value
+  }
+
+  /// Gets the config value of a given source from the default namespace.
+  /// - Parameter key              Config key.
+  /// - Parameter source           Config value source.
+  @objc public func configValue(forKey key: String, source: RemoteConfigSource) ->
+    RemoteConfigValue {
+    guard !key.isEmpty else {
+      return RemoteConfigValue(data: Data(), source: .static)
+    }
+    let fullyQualifiedNamespace = self.fullyQualifiedNamespace(FIRNamespace)
+    var value: RemoteConfigValue!
+
+    queue.sync {
+      switch source {
+      case .remote:
+        value = configContent.activeConfig()[fullyQualifiedNamespace]?[key]
+      case .default:
+        value = configContent.defaultConfig()[fullyQualifiedNamespace]?[key]
+      case .static:
+        value = RemoteConfigValue(data: Data(), source: .static)
+      }
+    }
+    return value
+  }
+
+  @objc(allKeysFromSource:)
+  public func allKeys(from source: RemoteConfigSource) -> [String] {
+    var keys: [String] = []
+    queue.sync {
+      let fullyQualifiedNamespace = self.fullyQualifiedNamespace(FIRNamespace)
+      switch source {
+      case .default:
+        if let values = configContent.defaultConfig()[fullyQualifiedNamespace] {
+          keys = Array(values.keys)
+        }
+      case .remote:
+        if let values = configContent.activeConfig()[fullyQualifiedNamespace] {
+          keys = Array(values.keys)
+        }
+      case .static:
+        break
+      }
+    }
+    return keys
+  }
+
+  @objc public func keys(withPrefix prefix: String?) -> Set<String> {
+    var keys = Set<String>()
+    queue.sync {
+      let fullyQualifiedNamespace = self.fullyQualifiedNamespace(FIRNamespace)
+
+      if let config = configContent.activeConfig()[fullyQualifiedNamespace] {
+        if let prefix = prefix, !prefix.isEmpty {
+          keys = Set(config.keys.filter { $0.hasPrefix(prefix) })
+        } else {
+          keys = Set(config.keys)
+        }
+      }
+    }
+    return keys
+  }
+
+  public func countByEnumerating(with state: UnsafeMutablePointer<NSFastEnumerationState>,
+                                 objects buffer: AutoreleasingUnsafeMutablePointer<AnyObject?>,
+                                 count len: Int) -> Int {
+    var count = 0
+    queue.sync {
+      let fullyQualifiedNamespace = self.fullyQualifiedNamespace(FIRNamespace)
+
+      if let config = configContent.activeConfig()[fullyQualifiedNamespace] as? NSDictionary {
+        count = config.countByEnumerating(with: state, objects: buffer, count: len)
+      }
+    }
+    return count
```

Comment:
Probably could return the value from the sync block, rather than passing it through the context.